JSON_mapping_plugin/src/map_types/plugin_mapping.cpp

```diff
@@ -65,31 +102,155 @@ int PluginMapping::call_plugins(const MapArguments& arguments) const {
         return err;
     } // Return 1 if no request receieved
 
-    err = callPlugin(arguments.m_interface->pluginList, request_str.c_str(), arguments.m_interface);
-    if (err) {
-        // add check of int udaNumErrors() and if more than one, don't wipe
-        // 220 situation when UDA tries to get data and cannot find it
-        if (err == 220)
-            closeUdaError();
-        return err;
-    } // return code if failure, no need to proceed
+   
+    /*
+     *
+     * generate subset info then remove subset syntax from
+     *  request string
+     *
+     */
 
-    if (m_plugin == "UDA" && arguments.m_sig_type == SignalType::TIME) {
-        // Opportunity to handle time differently
-        // Return time SignalType early, no need to scale/offset
-        err = imas_json_plugin::uda_helpers::setReturnTimeArray(arguments.m_interface->data_block);
-        return err;
+    
+    IDAM_PLUGIN_INTERFACE* plugin_interface = arguments.m_interface;
+    REQUEST_DATA request = *plugin_interface->request_data;
+
+    request.source[0] = '\0';
+    strcpy(request.signal, request_str.c_str());
+    makeRequestData(&request, *plugin_interface->pluginList, plugin_interface->environment);
+    SUBSET datasubset = request.datasubset;
+    subset::log_request_status(&request, "request block before interception: ");
+
+
+    // assume subsetting is requested if the final part of the request string is
+    // in sqaure bracktes
+    if (request_str.back() == ']' and request_str.rfind('[')!=std::string::npos)
```

Comment:
SLICE is optional so can check if slice.has_value() here
We briefly discussed separating the request string and the slice string separately so don't have to manually erase from the request string like this